apps/api/test/functional/provider-earnings.spec.ts

```diff
@@ -0,0 +1,84 @@
+import type { Provider } from "@akashnetwork/database/dbSchemas/akash";
+import { format, subDays } from "date-fns";
+import mcache from "memory-cache";
+import nock from "nock";
+
+import { app, initDb } from "@src/app";
+import { closeConnections } from "@src/db/dbConnection";
+import type { ProviderEarningsResponse } from "@src/provider/http-schemas/provider-earnings.schema";
+
+import { createAkashBlock, createProvider } from "@test/seeders";
+
+describe("Provider Earnings API", () => {
+  let provider: Provider;
+
+  beforeAll(async () => {
+    await initDb();
+
+    provider = await createProvider();
+    await Promise.all([
+      createAkashBlock({
+        datetime: new Date().toISOString(),
+        isProcessed: true,
+        totalUUsdSpent: 1000,
+        height: 300
+      }),
+      createAkashBlock({
+        datetime: subDays(Date.now(), 1).toISOString(),
+        height: 200
+      }),
+      createAkashBlock({
+        datetime: subDays(Date.now(), 2).toISOString(),
+        height: 100
+      })
+    ]);
+  });
+
+  afterAll(async () => {
+    await closeConnections();
+    mcache.clear();
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  describe("GET /internal/provider-earnings/{owner}?from=YYYY-MM-DD&to=YYYY-MM-DD", () => {
+    it("should return earnings for a valid provider and date range", async () => {
+      const from = format(subDays(Date.now(), 2), "yyyy-MM-dd");
+      const to = format(subDays(Date.now(), 1), "yyyy-MM-dd");
+
+      const response = await app.request(`/internal/provider-earnings/${provider.owner}?from=${from}&to=${to}`);
+
+      const data = (await response.json()) as ProviderEarningsResponse;
+
+      expect(response.status).toBe(200);
+      expect(data).toHaveProperty("earnings");
+      expect(data.earnings).toHaveProperty("totalUAktEarned");
+      expect(data.earnings).toHaveProperty("totalUUsdcEarned");
+      expect(data.earnings).toHaveProperty("totalUUsdEarned");
+    });
+
+    it("should return 404 for an invalid provider", async () => {
+      const nonExistentOwner = "0x1234567890abcdef1234567890abcdef12345678";
+      const from = format(subDays(Date.now(), 2), "yyyy-MM-dd");
+      const to = format(subDays(Date.now(), 1), "yyyy-MM-dd");
+
+      const res = await app.request(`/internal/provider-earnings/${nonExistentOwner}?from=${from}&to=${to}`);
```

Comment:
comment(blocking): please remove all reference to the `/internal/` route and use `/v1`